### DISCUSSION TITLE
FunctionValuePostprocessor to printout the values read by PiecewiseMultilinear

### DISCUSSION THREAD

**Discussion by styyokuda** [idaholab/moose#29625] [Q&A General]
### Check these boxes if you have followed the posting rules.

- [X] Q&A General is the most appropriate section for my question
- [X] I have consulted the posting Guidelines on the Discussions front page
- [X] I have searched the Discussions forum and my question has not been asked before
- [X] I have searched the MOOSE website and the documentation does not answer my question
- [X] I have formatted my post following the posting guidelines (screenshots as a last resort, triple back quotes around pasted text)

### Question

All,
I am using FunctionValuePostprocessor in Postprocessors to printout the values of a file read by PiecewiseMultilinear in Functions but it is seemingly not working (I got just zeros).
Please help. S.Thomas

**Comment by GiudGiud** [comment]
Did you check the execute_on parameter? Is it being executed often enough?

**Reply by styyokuda** [reply]
Thank you, GiudGiud!
I set as 
execute_on = 'initial timestep_end'

**Reply by styyokuda** [reply]
The file read by PiecewiseMultilinear in Functions has a form of
AXIS Y
xxx xxx xxx .... xxx
AXIS T
yyy yyy yyy ... yyy
DATA
zzz zzz zzz ... zzz
.
.
zzz zzz zzz ... zzz

So, we may have to sum the values in DATA for AXIS Y at each time step?
FunctionElementIntegral may help?

**Reply by GiudGiud** [reply]
the FunctionValuePP is only computing one value of the function, at the origin (x=0, y=0, z=0) by default
if you want an integral then I agree FunctionElementIntegral does it
